src/main/kotlin/com/sourcegraph/cody/agent/protocol/AutocompleteParams.kt

```diff
@@ -1,15 +1,13 @@
 package com.sourcegraph.cody.agent.protocol
 
-import com.google.gson.annotations.SerializedName
-
-enum class AutocompleteTriggerKind {
-  @SerializedName("Automatic") AUTOMATIC,
-  @SerializedName("Invoke") INVOKE,
+enum class AutocompleteTriggerKind(val value: String) {
+  AUTOMATIC("Automatic"),
+  INVOKE("Invoke"),
 }
 
 data class AutocompleteParams(
     val filePath: String,
     val position: Position,
-    val triggerKind: AutocompleteTriggerKind? = AutocompleteTriggerKind.AUTOMATIC,
+    val triggerKind: String? = AutocompleteTriggerKind.AUTOMATIC.value,
     val selectedCompletionInfo: SelectedCompletionInfo? = null
 )
```

Comment:
Are you sure `@SerializedName` doesnt work? I had tested it originally as part of https://github.com/sourcegraph/jetbrains/pull/88/ and it appeared to work fine :thinking: https://stackoverflow.com/questions/16740078/serialize-and-deserialize-enum-with-gson

Reply:
I created a sample project with the following code:
```kotlin
import com.google.gson.Gson
import com.google.gson.annotations.SerializedName

enum class AutocompleteTriggerKind {
    @SerializedName("Automatic") AUTOMATIC,
    @SerializedName("Invoke") INVOKE,
  }

data class User(val firstName: String, val lastName: String, val kind: AutocompleteTriggerKind) {
    val banana: String = ""
}

fun main() {
    val gson = Gson().toJson(User("Tom", "Broody", AutocompleteTriggerKind.AUTOMATIC))
    println(gson)
}
```
which yields the following output
`{"firstName":"Tom","lastName":"Broody","kind":"Automatic","banana":""}`

Reply:
It isn't mapped correctly on the agent side. It was always converted to value `0` idk why

Reply:
![Screenshot 2023-11-15 at 15 56 45](https://github.com/sourcegraph/jetbrains/assets/9321940/3029e278-6cbd-4df7-af97-39dc2795b573)

https://github.com/sourcegraph/jetbrains/assets/9321940/c72c79c4-7988-49fb-b1a6-38a5f0b5f16f

Reply:
This is on main branch

Reply:
Looks like this is the issue :facepalm: https://sourcegraph.com/github.com/eclipse-lsp4j/lsp4j/-/blob/org.eclipse.lsp4j.jsonrpc/src/main/java/org/eclipse/lsp4j/jsonrpc/json/adapters/EnumTypeAdapter.java?L53-77

Reply:
Hm, but the serialization on the client side seems to work fine.

I tested it with `SourcegraphApiContentHelper.toJson` (I believe it uses the object mapper that is actually used).

It still can be either an issue with the deserialization on the agent side or some issue that is specific to the logs
 only.

Reply:
lsp4j.jsonrpc uses a different Gson builder with different typadapters set afaik